### PR TITLE
Implemented: functionality to display 'Shopify' related information for a product variant on product inventory detail (#276nvc1).

### PR DESCRIPTION
**Pull request by meet-aniket**
Dependent on #73 
- Make data dynamic on the product inventory detail page.

image

### CODE REVIEW

File: src/store/modules/util/actions.ts [hunk 1]
```diff
@@ -26,6 +26,120 @@ const actions: ActionTree<UtilState, RootState> = {
     } catch (err) {
       console.error("error", err);
     }
+  },
+
+  // Get shopify enumeration description
+  async getShopifyEnumerations({ state, commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts]:
```suggestion
  async getEnumerations({ state, commit }, payload) {
```

File: src/store/modules/util/actions.ts [hunk 2]
```diff
@@ -26,6 +26,120 @@ const actions: ActionTree<UtilState, RootState> = {
...
+    const enumIds = payload.map((enumeration: any) => enumeration.productIdentifierEnumId)
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts]: I think we could have redundant values

Comment (adityasharma7) [inline src/store/modules/util/actions.ts]:
```suggestion
    const enumIds = payload.map((productStore: any) => productStore.productIdentifierEnumId)
```

File: src/store/modules/util/actions.ts [hunk 3]
```diff
@@ -26,6 +26,120 @@ const actions: ActionTree<UtilState, RootState> = {
...
+    const currentEnums = JSON.parse(JSON.stringify(state.enumerations));
+    const currentEnumIds = Object.keys(state.enumerations);
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts]: I think this is not needed

File: src/store/modules/util/actions.ts [hunk 4]
```diff
@@ -26,6 +26,120 @@ const actions: ActionTree<UtilState, RootState> = {
...
+
+    const enumIdFilter = enumIds.reduce((enums: any, enumId: any) => {
+      if(!currentEnumIds.includes(enumId)) {
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts]: We could directly check if there is any value for key in currentEnums

File: src/store/modules/util/actions.ts [hunk 5]
```diff
@@ -26,6 +26,120 @@ const actions: ActionTree<UtilState, RootState> = {
...
+  },
+
+  // Get shopify configIds
+  async getShopifyConfigIds({ state, commit, dispatch }, payload) {
+    const prdtStoreIds = payload.shopifyProductStores.map((prdtStore: any) => prdtStore.productStoreId)
+    const currentConfigs = JSON.parse(JSON.stringify(state.shopifyConfigs));
+    const currentConfigIds = Object.keys(state.shopifyConfigs);
+
+    const prdtStoreFilter = prdtStoreIds.reduce((stores: any, storeId: any) => {
+      if(!currentConfigIds.includes(storeId)) {
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts]: We could directly check if there is any value for key in currentConfigs

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/util/actions.ts (hunk 1 above)
Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Done Sir.

File: src/store/modules/util/actions.ts (hunk 2 above)
Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Sir, updated code.

Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Done Sir.

File: src/store/modules/util/actions.ts (hunk 3 above)
Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Sir, changed code.

File: src/store/modules/util/actions.ts (hunk 4 above)
Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Sir, improved code.

File: src/store/modules/util/actions.ts (hunk 5 above)
Comment (meet-aniket) [reply src/store/modules/util/actions.ts]: Sir, improved code.

**Comment by Dhiraj1405** [comment]
Not priority for now. Hence, closing this pull request.

**Pull request closed without merging**